ehr_ml/clmbr/dataset.py

```diff
@@ -56,11 +56,30 @@ def prepare_batch_thread(
         result = finalize_data(item, device)
         out_queue.put(result)
 
+def convert_pid(pid: int, search_list, result_list) -> Tuple[int, int]:
+    pid_index = bisect.bisect_left(search_list, pid)
+    assert (
+        search_list[pid_index] == pid
+    ), f"patient ID {pid} not in timeline"
+    return_pid = result_list[pid_index]
+    return return_pid
 
+def orig2ehr_pid(orig_pid: int, extract_dir: str):
+    timelines = timeline.TimelineReader(os.path.join(extract_dir, "extract.db"))
+    all_original_pids = timelines.get_original_patient_ids()
```

Comment:
This is going to be very slow. Loading a whole TImelineReader to convert a single patient id.

Reply:
Please simply take the reader as an argument.

Reply:
Should be resolved in the latest commit, thanks for the comments!